src/main/java/SearchEngine.java

```diff
@@ -0,0 +1,109 @@
+import impl.AllStrategyImpl;
+import impl.AnyStrategyImpl;
+import impl.NoneStrategyImpl;
+import strategies.SearchStrategy;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.stream.Stream;
+
+public class SearchEngine {
+    Scanner scanner = new Scanner(System.in);
+    ReadManager readManager;
+
+    public SearchEngine(ReadManager readManager) {
+        this.readManager = readManager;
+    }
+
+    public void run() {
+        ArrayList<String> peopleList = readManager.readFile();
```

Comment:
The same here. Please use `List<String>` instead of `ArrayList<String>`. Methods also should return interface as data type.
Why? If somebody changes realization to `LinkedList<String>` in the `readFile()` method, you will need to fix it everywhere.

Reply:
fixed